utils/response.go

```diff
@@ -48,3 +49,27 @@ func ResponseWasStatusCode(resp autorest.Response, statusCode int) bool { // nol
 
 	return false
 }
+
+// ResponseWasBadRequestWithNotRegistered is a workaround for https://github.com/Azure/azure-rest-api-specs/issues/12759
+func ResponseWasBadRequestWithNotRegistered(resp autorest.Response, err error) bool {
```

Comment:
Whilst these workarounds are relatively rare, they do crop up from time to time in services, would it be worth making this function more utilitarian and provide the `serviceCode` as a parameter for explicit matching? e.g.
```suggestion
func ResponseWasBadRequestWithServiceCode(resp autorest.Response, err error, serviceCode string) bool {
```
Otherwise, I think this should be part of the service package itself, not the `utils` package since it's a single use-case?

Reply:
I have found some [similar usage](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/appservice/function_app_function_resource.go#L205) of `ResponseWasBadRequest` in `app_function`, so it maybe a common issue in resources.. but making it accept `serviceCode` as parameter is a good idea, thank you.